### PR TITLE
Update plugin-core to version 0.12.0

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/composables/events/PluginEvent.js [hunk 1]
```diff
@@ -31,6 +31,13 @@ const UpdateEvent = new Subject();
  */
 const ParseEvent = new Subject();
 
+/**
+ * Represent a rxjs Event object to emit and to receive events about plugins component rendering.
+ * Subject contains an FileInput[] of all modified files.
```
Comment (justinehell) [inline src/composables/events/PluginEvent.js:36]: `a FileInput[]`

File: src/components/drawer/ComponentDetailPanel.vue [hunk 2]
```diff
@@ -169,12 +169,14 @@ let viewSwitchSubscription;
 /**
  * Update local component data and emit DrawEvent & RenderEvent events.
  */
-function save() {
+async function save() {
```
Comment (justinehell) [inline src/components/drawer/ComponentDetailPanel.vue:172]: no await ?

File: src/components/editor/MonacoEditor.vue [hunk 3]
```diff
@@ -100,11 +102,24 @@ async function updateEditorContent() {
   editor.setValue(value);
 }
 
+/**
+ * update editor value from files.
```
Comment (justinehell) [inline src/components/editor/MonacoEditor.vue:112]: `Update`

File: tests/unit/components/editor/MonacoEditor.spec.js [hunk 4]
```diff
@@ -151,6 +152,25 @@ describe('Tess component: MonacoEditor', () => {
...
   });
 
+  describe('Test function: updateEditorContentFromFiles', () => {
+    it('should call not call setValue without files', async () => {
+      await wrapper.vm.updateEditorContentFromFiles([]);
+
+      expect(setValue).not.toHaveBeenCalled();
+    });
+
+    it('should call not call setValue with valid file', async () => {
```
Comment (justinehell) [inline tests/unit/components/editor/MonacoEditor.spec.js:162]: remove `not call`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: cypress/e2e/ModelizerPage/TextView/file/update.file.feature [hunk 5]
```diff
@@ -92,7 +92,8 @@ Feature: Test modelizer text view: update file's content
 
     When I set active file content to "updated content"
     And  I click on "[data-cy=\"file-explorer\"] [data-cy=\"file-explorer-buttons-branch.txt\"]"
+    And I wait 1 seconds
```
Comment (justinehell) [inline cypress/e2e/ModelizerPage/TextView/file/update.file.feature:95]: `second`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: tests/unit/composables/PluginManager.spec.js [hunk 6]
```diff
@@ -139,4 +158,18 @@ describe('Test composable: PluginManager', () => {
       expect(PluginManager.isParsableFile('parsable')).toEqual(true);
     });
   });
+
+  describe('Test function: renderPlugin', () => {
+    it('should return updated file and emit a RenderEvent', async () => {
+      await PluginManager.initPlugins();
+      const plugin = PluginManager.getPluginByName('test');
```
Comment (k-y0u) [inline tests/unit/composables/PluginManager.spec.js:165]: jump line before and after

File: src/components/editor/MonacoEditor.vue [hunk 7]
```diff
@@ -85,7 +86,12 @@ async function createEditor() {
  * Needed when viewType is firstly set to 'model' and ModelizerTextView is hidden,
  * meaning 'container' height and width are set to 0.
  */
-function updateEditorLayout() {
+async function updateEditorLayout() {
+  // For some unknowns reasons, editor is null, only appear on cypress test.
```
Comment (k-y0u) [inline src/components/editor/MonacoEditor.vue:90]: unknowns -> no S
`but only appears on cypress tests`

File: src/composables/events/PluginEvent.js (hunk 1 above)
Comment (Zorin95670) [reply src/composables/events/PluginEvent.js:36]: done

File: src/components/editor/MonacoEditor.vue (hunk 3 above)
Comment (Zorin95670) [reply src/components/editor/MonacoEditor.vue:112]: done

File: tests/unit/components/editor/MonacoEditor.spec.js (hunk 4 above)
Comment (Zorin95670) [reply tests/unit/components/editor/MonacoEditor.spec.js:162]: done

File: tests/unit/components/editor/MonacoEditor.spec.js [hunk 8]
```diff
@@ -151,6 +152,25 @@ describe('Tess component: MonacoEditor', () => {
     });
   });
 
+  describe('Test function: updateEditorContentFromFiles', () => {
+    it('should call not call setValue without files', async () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/components/editor/MonacoEditor.spec.js:156]: done

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/composables/events/PluginEvent.js [hunk 9]
```diff
@@ -31,6 +31,13 @@ const UpdateEvent = new Subject();
  */
 const ParseEvent = new Subject();
 
+/**
+ * Represent a rxjs Event object to emit and to receive events about plugins component rendering.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/events/PluginEvent.js:35]: done

File: cypress/e2e/ModelizerPage/TextView/file/update.file.feature (hunk 5 above)
Comment (Zorin95670) [reply cypress/e2e/ModelizerPage/TextView/file/update.file.feature:95]: done

File: src/composables/events/PluginEvent.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/events/PluginEvent.js:36]: done

File: src/components/editor/MonacoEditor.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/editor/MonacoEditor.vue:112]: done

File: src/components/drawer/ComponentDetailPanel.vue (hunk 2 above)
Comment (Zorin95670) [reply src/components/drawer/ComponentDetailPanel.vue:172]: done

File: src/components/editor/MonacoEditor.vue [hunk 10]
```diff
@@ -100,11 +102,24 @@ async function updateEditorContent() {
   editor.setValue(value);
 }
 
+/**
+ * update editor value from files.
+ * @param {FileInput[]} files - All updated files.
+ * @return {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+async function updateEditorContentFromFiles(files) {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/editor/MonacoEditor.vue:116]: done

File: src/composables/PluginManager.js [hunk 11]
```diff
@@ -117,3 +119,28 @@ export function getPluginByName(name) {
 export function isParsableFile(file) {
   return instanciatePlugins.some((plugin) => plugin.isParsable(file));
 }
+
+/**
+ * Call render from selected plugin for specific project.
+ * Emit a RenderEvent.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/PluginManager.js:125]: done

File: src/composables/PluginManager.js [hunk 12]
```diff
@@ -117,3 +119,28 @@ export function getPluginByName(name) {
...
+ * an error.
+ */
+export async function renderPlugin(pluginName, projectId) {
+  const plugin = getPluginByName(pluginName);
+  const config = new FileInput({
+    path: 'leto-modelizer.config.json',
+    content: '{}',
+  });
+  const files = await getProjectFiles(projectId);
+  const result = await Promise.allSettled(
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/PluginManager.js:138]: done

File: tests/unit/composables/PluginManager.spec.js (hunk 6 above)
Comment (Zorin95670) [reply tests/unit/composables/PluginManager.spec.js:165]: done

File: src/components/editor/MonacoEditor.vue (hunk 7 above)
Comment (Zorin95670) [reply src/components/editor/MonacoEditor.vue:90]: done

**Review by justinehell — DISMISSED** [review]

**Review by justinehell — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**